tf2_ros/src/buffer.cpp

```diff
@@ -44,10 +44,22 @@
 #define ROS_INFO printf
 #define ROS_WARN printf
 
-
 namespace tf2_ros
 {
 
+// Added to backport: https://github.com/ros2/geometry2/pull/281
+std::mutex g_object_map_to_cb_handle_mutex;
+std::map<void*, std::unordered_map<TimerHandle, tf2::TransformableCallbackHandle>> g_object_map_to_cb_handle;
+
+void deleteTransfromCallbackHandle(void *class_ptr, const TimerHandle &timer_handle)
+{
+  g_object_map_to_cb_handle[class_ptr].erase(timer_handle);
```

Comment:
Is it ever the case that this can be called without the class pointer key first having been added?  It may make sense to use `g_object_map_to_cb_handle.at(class_ptr)` here instead (it can throw an exception, but at least it won't corrupt memory then).

The same goes for all uses of `operator[]` below; it bears thinking about whether we should use `at` instead.

Reply:
Should not be possible, since all calls of delete are done after it has been created first. But I agree with you that `at` is an better idea.

Reply:
I think I *slightly* prefer `insert()`, but I really don't have a strong opinion.